utilities/env_librados.cc

```diff
@@ -0,0 +1,1486 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "rocksdb/utilities/env_librados.h"
+#include "util/random.h"
+#include <mutex>
+
+/* GLOBAL DIFINE */
+// #define DEBUG
+#ifdef DEBUG
+#include <cstdio>
+#include <sys/syscall.h>
+#include <unistd.h>
+#define LOG_DEBUG(fmt, ...)  do{\
+    printf("[%ld:%s:%i:%s]", syscall(SYS_gettid), __FILE__, __LINE__, __FUNCTION__);\
+    printf(fmt, __VA_ARGS__);\
+  }while(0)
+#else
+#define LOG_DEBUG(fmt, ...)
+#endif
+
+template<typename T1, typename T2> bool min(T1 a, T2 b) {return a > b ? b : a;}
+
+/* GLOBAL CONSTANT */
+// maximum dir/file that can store in the fs
+const int MAX_ITEMS_IN_FS = 1 << 30;
+// root dir tag
+const std::string ROOT_DIR_KEY = "/";
+/*
+ * FID format:
+ * |<------------ DB id ------------>|<------------ file id ---------->|
+ *  0000000000000000-0000000000000000.0000000000000000-0000000000000000
```

Comment:
do we actually need to constrain the format of db_id?  Couldn't it just be a string, and everythign else is $dbid (directory) or  $dbid.$fid ?